### PR TITLE
Feat/removal challenge justification display

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
- Adds display for Justification for Removal and Challenge Requests
- Adds a new field `externalDisputeID` in `Request` entity, to make it easier to link it to evidences
- Adds `challengeTime` field in `Request` enitity
- Updates the addresses in List and Item Cards

---

## PR-Codex overview
This PR enhances the Curate application by adding challenge time tracking, improving UI components, updating subgraph queries, and refining data fetching functions.

### Detailed summary
- Added `challengeTime` tracking to requests
- Updated UI components for `Header`, `RegistryInformationCard`, and `JustificationDetails`
- Modified subgraph queries for `useRequestsQuery` and `useEvidences`
- Implemented `fetchJsonIpfs` data fetching function
- Updated environment variables and dependencies
- Enhanced scrollbar styling
- Updated schema with `externalDisputeID` field

> The following files were skipped due to too many changes: `web/src/components/HistoryDisplay/Party/JustificationDetails.tsx`, `web/src/components/HistoryDisplay/Party/index.tsx`, `web/src/components/InformationCards/RegistryInformationCard/TopInfo.tsx`, `web/src/components/InformationCards/ItemInformationCard/index.tsx`, `web/src/components/HistoryDisplay/index.tsx`, `web/src/components/HistoryDisplay/Party/JustificationModal.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Summary by CodeRabbit

- **New Features**
  - Introduced `externalDisputeID` and `challengeTime` fields in request details.
  - Added new React components: `Party`, `JustificationDetails`, and `JustificationModal`.
  - Introduced custom hook `useEvidences` for fetching evidence data.
  - Implemented a new function `fetchJsonIpfs` for fetching JSON data from IPFS.

- **Enhancements**
  - Updated several components to improve alias display and justification handling.
  - Updated `@kleros/ui-components-library` to version `2.13.1`.
  - Enhanced styling for custom scrollbars.

- **Configuration**
  - Updated subgraph URLs in `.env.devnet.public`.

- **Bug Fixes**
  - Fixed `fetcher` function in `GraphqlBatcher.tsx` to handle core service requests.

### CODE REVIEW

File: web/src/components/HistoryDisplay/Party/JustificationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      setIsLoadingJustification(false);
+      return;
+    }
+
+    fetchJsonIpfs(getIpfsUrl(uri))
+      .then((res) => {
+        setJustification(res as Justification);
+      })
+      .finally(() => setIsLoadingJustification(false));
+  }, [evidenceData, isRemoval, request]);
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/components/HistoryDisplay/Party/JustificationModal.tsx:70]: handled inside fetchJsonIpfs function itself

**Comment by jaybuidl** [comment]
image

1) Let's make sure that the copied address and the display address is the same

2) Let's make the address label a link going to the explorer + remove the explorer icon

**Review by kemuru — COMMENTED** [review]
just 2 tiny things:

1) remove the space between by and :
image

2) put `text-decoration: underline` if you hover this address
image

**Review by kemuru — APPROVED** [review]
lgtm

**Pull request merged**